### PR TITLE
Part II: Query multiple tables

### PR DESCRIPTION
**Pull request by haroldmud**
## In this branch I did the following: :point_down: 

- Created a table named owners with the given columns.
- Created a table named owners with the given columns.
- Modified animals table.
- Inserted the given data into the `owner` table.
InsertOwner
- Inserted the given data into the species table
$InsertSpecies
- Modified my inserted animals so it includes the species_id value and owner_id ones.
$AddOwners
- Wrote queries (using JOIN) to answer the following questions.
  - What animals belong to Melody Pond?
$MelodyJoin
  - List of all animals that are pokemon (their type is Pokemon).
Screenshot from 2022-07-09 00-46-40
  - List all owners and their animals, remember to include those that don't own any animal.
$ownersJoinAnimals
  - How many animals are there per species?
Screenshot from 2022-07-09 00-35-59
  - List all Digimon owned by Jennifer Orwell.
$DigimonJoinJennifer
  - List all animals owned by Dean Winchester that haven't tried to escape.
 $DeanJoinNoEscape
  - Who owns the most animals?
 $JoinMostAnimalOwner

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
 CREATE TABLE animals ( ID int, Name varchar(100), Date_of_birth date, Escape_attempts int, Neutered boolean, Weight_kg decimal );
+
+ALTER TABLE animals DROP COLUMN ID; -----> deletes a specific column(ID)
+ALTER TABLE animals ADD ID SERIAL PRIMARY KEY;
+
 ALTER TABLE animals ADD Species varchar(100);
+ALTER TABLE animals DROP COLUMN species;
+ALTER TABLE animals ADD COLUMN species_id varchar(100);
+
+ALTER TABLE animals ADD COLUMN owner_id varchar(100);
```
Comment (gtekle) [inline schema.sql]: - [x] The `species_id` and `owner_id` are foreign keys for the `species` table and `owenrs` table respectively. Therefore, you should use a syntax like the following. You may also check [this](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-foreign-key/) page for further reading.
```
ALTER TABLE child_table 
ADD CONSTRAINT constraint_name 
FOREIGN KEY (fk_columns) 
REFERENCES parent_table (parent_key_columns);
```

File: data.sql [hunk 2]
```diff
@@ -30,3 +31,74 @@ VALUES ( 10, 'Blossom', '1998-10-13', 3, 'true', '17', '' );
...
+INSERT INTO species ( ID, Name )
+VALUES ( 1, 'Pokemon' );
+
+INSERT INTO species ( ID, Name )
+VALUES ( 2, 'Digimon' );
+
+
+----INNER JOIN----
+
+SELECT owner.full_name, animals.name FROM animals INNER JOIN owner ON owner.full_name = animals.owner_id WHERE owner.full_name='Melody Pond';
```
Comment (gtekle) [inline data.sql:102]: - [x] You should use parallel columns when you are using JOINS. That is, we can't compare an ID with a NAME. It should be compared ID vs ID. Therefore, the correct SELECT statement with INNER JOIN would be: (**owner.id**)
`SELECT owner.full_name, animals.name FROM animals INNER JOIN owner ON owner.id = animals.owner_id WHERE owner.full_name='Melody Pond';
`

File: data.sql [hunk 3]
```diff
@@ -30,3 +31,74 @@ VALUES ( 10, 'Blossom', '1998-10-13', 3, 'true', '17', '' );
...
+SELECT species.name, animals.name FROM animals INNER JOIN species ON species.name = animals.species_id WHERE species.name='Pokemon';
```
Comment (gtekle) [inline data.sql]: - [x] You should use **species.id = animals.species_id** instead for the same reason mentioned in the previous comment.

File: data.sql [hunk 4]
```diff
@@ -30,3 +31,74 @@ VALUES ( 10, 'Blossom', '1998-10-13', 3, 'true', '17', '' );
...
+SELECT owner.full_name, animals.name FROM animals INNER JOIN owner ON owner.full_name = animals.owner_id;
+SELECT COUNT(*) FROM animals WHERE species_id = 'Pokemon';
```
Comment (gtekle) [inline data.sql:103]: - [ ] If it is done correctly, `species_id` has a data type of `INT` though it is assigned a `String` value of `Pokeman`. What you need to do achieve this selection is to make a left join between the `animals`  and species table based on the **animals.species_id = species.id** and then check if **species.name = 'Pokemon'**.

**Review by gtekle — CHANGES_REQUESTED** [review]
Hi @haroldmud :wave: , 

# STATUS: Changes Requested :red_circle: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-there-01


## Highlights
:heavy_check_mark: Descriptive PR summary
:heavy_check_mark: Good use of Gitflow
:heavy_check_mark: Professional ReadMe

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly check the inline comments below and I highly suggest that you go through the materials that discuss on how to create different types JOINs (INNER, LEFT, OUTER) and what their difference is.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by haroldmud** [comment]
Thank you for the review

**Comment by haroldmud** [comment]
Tried to use parallel COLUMN while using INNER JOIN for both owner_id and species_id, Implemented the Foreign key for the child tables...

**Pull request closed without merging**